arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Builders.kt

```diff
@@ -306,12 +289,12 @@ public class OptionRaise(private val raise: Raise<None>) : Raise<None> by raise
  */
 public class IorRaise<Error> @PublishedApi internal constructor(
   @PublishedApi internal val combineError: (Error, Error) -> Error,
-  private val state: Atomic<Option<Error>>,
+  private val state: Atomic<Any?>,
   private val raise: Raise<Error>,
-) : Raise<Error> {
-
-  @RaiseDSL
-  override fun raise(r: Error): Nothing = raise.raise(combine(r))
+) : Raise<Error> by raise {
```

Comment:
I don't think we can remove the override for `raise`, right? Don't we otherwise `combine` the error when doing `raise` now, are we? 🤔

I think a test is missing here:

```kotlin
ior(String::plus) {
  Ior.Both("Hello", Unit).bind()
  raise("World")
} shouldBe Ior.Left("Hello World")
```

Reply:
I have the final combine call happening inside the `ior` builder itself. This is so that try catch can recover from the error, and also so that any races in calling `raise` is only won by the call that actually reaches the builder, which is the behaviour of the rest of the builders.
That test passes (after adding a missing space) so no issues there.

removing the raise override is also good because it makes transitioning to context receivers much easier.